streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -76,6 +172,29 @@ object ZStream extends Serializable {
   def apply[R, E, M, B, A](process: ZManaged[R, Nothing, Control[R, E, M, B, A]]): ZStream[R, E, M, B, A] =
     new ZStream(process)
 
+  /**
+   * Creates a stream from a [[zio.Chunk]] of values
+   *
+   * @tparam A the value type
+   * @param c a chunk of values
+   * @return a finite stream of values
+   */
+  def fromChunk[A](c: => Chunk[A]): ZStream[Any, Nothing, Any, Unit, A] =
+    ZStream {
+      Managed {
+        Ref.make(0).map { iRef =>
+          val l = c.length
+          val pull = iRef.get.flatMap { i =>
+            if (i >= l)
+              Pull.endUnit
+            else
+              iRef.update(_ + 1) *> Pull.emit(c(i))
+          }
+          Reservation(UIO.succeedNow(Control(pull, Command.noop)), _ => UIO.unit)
+        }
+      }
```

Comment:
`Managed.succeed`?

Reply:
Isn't it necessary to allocate the `Ref` outside of the reservation?

Reply:
I don't think so, check the implementation in `StreamEffect.scala`.

Reply:
ha yeah, great point 🤦‍♂

Reply:
but it has to be created outside of the `Control`, right? So perhaps `Managed.fromEffect` instead 🤔

Reply:
It's basically nitpicking at this point, but maybe this:
```scala
ZStream {
  for {
    ref  <- Ref.make(0).toManaged_
    len  = c.length
    pull = ref.get.flatMap { idx =>
      if (idx >= len) Pull.endUnit
      else ref.set(idx + 1) *> Pull.emit(c(idx))
    }
  } yield Control(pull, Command.noop)
}
```

Reply:
I have a slight preference for my version as it has less roundtrips but happy to be convinced otherwise 😊